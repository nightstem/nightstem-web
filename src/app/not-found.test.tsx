import { render, screen } from '@testing-library/react';
import { describe, expect, it, vi } from 'vitest';

import NotFoundPage, { metadata } from '@/app/not-found';

// Mock the NotFound component to isolate testing of not-found.tsx
vi.mock('@/screens/NotFound', () => ({
  NotFound: vi.fn(() => (
    <div data-testid="not-found-component">NotFound Component</div>
  )),
}));

describe(NotFoundPage, () => {
  it('renders correctly with snapshot', () => {
    const { container } = render(<NotFoundPage />);

    expect(container.firstChild).toMatchSnapshot();
  });

  it('renders NotFound component', () => {
    render(<NotFoundPage />);

    const notFoundComponent = screen.getByTestId('not-found-component');

    expect(notFoundComponent).toBeInTheDocument();
    expect(notFoundComponent).toHaveTextContent('NotFound Component');
  });

  it('has correct displayName', () => {
    expect(NotFoundPage.displayName).toBe('NotFoundPage');
  });

  describe('metadata', () => {
    it('exports correct metadata object', () => {
      expect(metadata).toBeDefined();
      expect(metadata).toStrictEqual({
        title: 'Not found | Nightstem',
        robots: 'noindex',
      });
    });

    it('has correct title for SEO', () => {
      expect(metadata.title).toBe('Not found | Nightstem');
    });

    it('has noindex robots directive', () => {
      expect(metadata.robots).toBe('noindex');
    });
  });
});
