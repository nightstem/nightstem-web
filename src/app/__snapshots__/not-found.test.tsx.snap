// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`NotFoundPage > renders correctly with snapshot 1`] = `
<div
  class="flex flex-col h-dvh w-dvw"
>
  <nav
    class="w-full flex justify-center shadow-sm shadow-neutral-500"
  >
    <div
      class="screen-container px-2 py-2.5 flex items-center justify-between"
    >
      <div
        class="inline-flex items-center gap-2"
      >
        <img
          alt="Nightstem icon"
          class="object-contain h-6 w-6"
        />
        <span
          class="heading-md"
        >
          Nightstem
        </span>
      </div>
      <ol>
        <li>
          <a
            class="select-none whitespace-nowrap font-medium cursor-pointer disabled:cursor-not-allowed transition-colors duration-200 motion-reduce:transition-none focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-offset-2 ring-offset-dark text-base leading-[1.15] px-4 py-2 rounded-md text-neutral-400 hover:text-neutral-300 active:text-neutral-300 focus-visible:ring-neutral-400 disabled:opacity-60 disabled:hover:text-neutral-400 disabled:hover:bg-transparent disabled:active:bg-transparent"
            href="/"
          >
            Home
          </a>
        </li>
      </ol>
    </div>
  </nav>
  <main
    class="screen-container h-full max-h-full overflow-auto"
  >
    <div
      data-testid="not-found-component"
    >
      NotFound Component
    </div>
  </main>
</div>
`;
